craft_providers/multipass/multipass_provider.py

```diff
@@ -170,7 +176,7 @@ def ensure_provider_is_available(cls) -> None:
             install()
         ensure_multipass_is_ready()
 
-    def create_environment(self, *, instance_name: str) -> Executor:
+    def create_environment(self, *, instance_name: str) -> "Executor":
```

Comment:
nit: if you do a `from __future__ import annotations`, then you don't have to do this legacy practice of quoting types.